### PR TITLE
Support for using http/s ws/s protocols

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by cristianonicolai** [comment]
@ricardozanini fyi

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@mswiderski we also have to decide if the operator will append `/graphql` to the URL or not.
IMO, we shouldn't since the data index url could have other contexts that can be used with this variable.

The application should not add the constant `http` in the beggining of the URL since only the infra knows if the connection is secure or not. Example: the route could be exposed only on secure manner. If it's the case, the application will stop working.

Bottom line is, I'd like to leave the operator with:

`KOGITO_DATAINDEX_URL=http://dataindex.ocp.apps.com`

or

`KOGITO_DATAINDEX_URL=https://dataindex.ocp.apps.com`

This way the app can figure out the scheme for secure/unsecure connections and add other protocols if need.

**Comment by cristianonicolai** [comment]
perhpas another variable to set the context where the Graphql service is available.

**Comment by mswiderski** [comment]
Few questions:
- so we really need to have context for graphql? Would it make sense to bind it to th riot context?
- how do we expect users to deal with it? Shall we inject to variables for data index - one with http and on with ws? That way we allow users to take the protocol they need directly from it

I am not big fun of these changes in the java code here even less if we would expect users to do similar magic ...

And I guess it’s time to create 06 version of the app and maybe move the UI (both travel agency and visas) to separate UI app so we could add scale to 0 story for the services and maybe auth...

**Comment by ricardozanini** [comment]
> so we really need to have context for graphql? Would it make sense to bind it to th riot context?

This question I should ask to you :). I don’t think the operator should inject the context in the url. This is not the operator domain. 

IMO, the best agnostic way to deal with this is having the operator inject the route name and another one flagging if it’s secure or not. Even protocol agnostic. 

Exemple:

KOGITO_DATAINDEX_ROUTE=dataindex.apps.com
KOGITO_DATAINDEX_ROUTE_SECURE=false

The application would append the protocol and the context of needed.

**Comment by ricardozanini** [comment]
Agree with the the version 6 of the application. Scale to 0 is not a domain of the operator right now. This is a Knative thing that we should add to the Roadmap, right?

**Comment by ricardozanini** [comment]
If you need the context, like the GraphQL service or UI, I’d add to another property not in the route. But I don’t think it should, IMO

**Comment by cristianonicolai** [comment]
My preference would be to keep the context _/graphql_ that is basically the standard for the API. 
Two separated variables one for http/https and ws/wss url, would be a more straightforward solution from the code perspective and variable definition, but I'm not completely against using a secure true/false.

**Comment by mswiderski** [comment]
I agree it would be better to have two variables that one will point to WS or HTTP (either secure or not) would be better.

so let's keep the graphql context path that is added by the app itself or injected by the operator? I'd say injected but open for other thoughts

**Comment by cristianonicolai** [comment]
@mswiderski that could be a variable with a default of /graphql, that way the operator could simply skip it, @ricardozanini wdyt?

**Comment by ricardozanini** [comment]
So the operator will:

- Inject a `KOGITO_DATAINDEX_HTTP_URL=http://dataindex.com/graphql`
- Inject a `KOGITO_DATAINDEX_WS_URL=ws://dataindex.com/graphql`
If secure, change the protocol to http**s** / ws**s**

?

Ok, I can live with that since the operator will know if the route is secure or not. But I'm not happy adding the context to the end of the URL, honestly. Although `graphql` is the standard for the service API we do have others contexts like `graphiql` and `metrics`, `health` and so on. Seems to me that we will lost context.

What about:

`KOGITO_DATAINDEX_GRAPHQL_HTTP_URL` and `KOGITO_DATAINDEX_GRAPHQL_WS_URL`?

This way we have context and others will understand better what story we want to tell.

**Comment by cristianonicolai** [comment]
@ricardozanini my suggestion is to not have the context in the variable, but add a third one.

- KOGITO_DATAINDEX_HTTP_URL=http://dataindex.com
- KOGITO_DATAINDEX_WS_URL=ws://dataindex.com
- KOGITO_DATAINDEX_GRAPHQL_CONTEXT=/graphql

This last one can be ignored as it will be the default anyway.

**Comment by ricardozanini** [comment]
Opened: https://issues.redhat.com/browse/KOGITO-720

Just a small note: I still don't think that we should add application context into operator code. :p
Won't cause a harm, though.

**Comment by cristianonicolai** [comment]
@ricardozanini  tbh I don't think the third parameter is necessary, I would be ok to only have the http and ws ones.

**Comment by ricardozanini** [comment]
@mswiderski ? Tomorrow I should work on KOGITO-720 since it's simple for 0.7.0-RC1

**Comment by mswiderski** [comment]
+1 on having dedicated variables one for http and one for ws. so pretty much what @cristianonicolai described above.

**Comment by ricardozanini** [comment]
We already have the HTTP one, I will rename it and add the WS. Problem is that you’re adding the scheme to the url in this project, and also expecting that the context is added to it. 

Let’s update this PR to append the context to the URL? While 0.7 isn’t released let’s use Nicolai workaround to guess the WS URL. 

Agreed?

**Comment by mswiderski** [comment]
so you mean to merge this PR now, right? If so fine with me

**Comment by cristianonicolai** [comment]
ok, let me update it, to append the context to the URL.

**Comment by cristianonicolai** [comment]
done

**Review by mswiderski — APPROVED** [review]
